### PR TITLE
ci: run tests and lint

### PR DESCRIPTION
**Pull request by dianwen**
Hopefully this doesn't cause too many conflicts w/ your workspace!

I recommend reviewing commits separately.

I commented on places where I changed the functionality of the code. Aside from that, I noticed lots of places were we catch errors but ignore them. Are we ok with that?

### CODE REVIEW

File: versions/v1/utils.ts [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
 
+import { getMetadata } from "../../shared/helpers/metadata";
+
 export const getFilePath = () => {
-    const cwd = process.cwd();
-    const metaData = getMetadata(path.join(cwd, '.satsuma.json'));
-}
+  const cwd = process.cwd();
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const metaData = getMetadata(path.join(cwd, ".satsuma.json"));
```
Comment (dianwen) [inline versions/v1/utils.ts:8]: This is not used so I disabled linting. Requires cleanup.

File: versions/v1/index.ts [hunk 2]
```diff
@@ -1,195 +1,235 @@
...
+    } catch (error) {
+      spinner.fail("Error uploading files");
+      process.exit(1);
+    }
+  },
+  validate: async (args) => {
+    validateFiles();
+    await validateExports();
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const deployKey = args.deployKey || getDeployKey(MD_PATH);
```
Comment (dianwen) [inline versions/v1/index.ts:84]: This is not used so I disabled linting. Requires cleanup.

File: versions/v1/index.ts [hunk 3]
```diff
@@ -1,195 +1,235 @@
...
+        databases,
+        graphql,
+        resolverFile,
+        typeDefsFile,
+        helpersFile,
+      };
+
+      const server = await createServer(config, typeDefs, resolvers, helpers);
+      // eslint-disable-next-line no-async-promise-executor
+      return new Promise(async () => {
```
Comment (dianwen) [inline versions/v1/index.ts:197]: Ignored this lint rule, but there are reasons for not using an async fn in the promise constructor. https://eslint.org/docs/latest/rules/no-async-promise-executor

File: versions/v1/index.ts [hunk 4]
```diff
@@ -1,195 +1,235 @@
...
+        console.log(`🍊Satsuma server listening at ${url}`);
+
+        const shutdownServer = () => {
+          console.log("Shutting down server...");
+          server
+            .stop()
+            .then(() => {
+              process.exit(0);
+            })
+            .catch(() => process.exit(1));
```
Comment (dianwen) [inline versions/v1/index.ts:208]: I added a catch to exit the process if there are rejections from stopping the server.

Seems like we dont care to resolve or reject this promise? It's a bit suss of a style.

**Review by androbwebb — APPROVED** [review]
Praise be!

Yeah, some of these issues need better fixes & cleaning, but this is great.

**Pull request merged**